lib/mllp/sender.ex

```diff
@@ -343,6 +314,48 @@ defmodule MLLP.Sender do
     stop_connection(state, reason, "process terminated")
   end
 
+  @should_reconnect_errors [
+    :closed,
+    :enotconn,
+    :econnreset,
+    :enetreset,
+    :econnaborted,
+    :econnrefused
+  ]
+  defp handle_send_error(state, error) when error in @should_reconnect_errors do
+    telemetry(
+      :status,
+      %{status: :disconnected, error: error, context: "send message failure"},
+      state
+    )
+
+    new_state = maintain_reconnect_timer(state)
+    reply = {:error, %{type: :send_failure, reason: error}}
+    {:reply, reply, new_state}
+  end
+
+  defp handle_send_error(state, error) do
+    # Add telemetry call here
+    reply = {:error, %{type: :send_failure, reason: error}}
+    {:reply, reply, state}
+  end
+
+  defp handle_recv_error(state, error) when error in @should_reconnect_errors do
+    telemetry(
+      :status,
+      %{status: :disconnected, error: error, context: "receive ACK failure"},
+      state
+    )
+
+    maintain_reconnect_timer(state)
+    {:error, %{type: :recv_failure, reason: error}}
+  end
+
+  defp handle_recv_error(_state, error) do
+    # Add telemetry call here
```

Comment:
Ditto. What do you think the  status for a telemetry call here should be? Before it was disconnected, but that's not going to be the case in any  case that hits this clause.

Reply:
same as above